lib/ddtrace/contrib/extensions.rb

```diff
@@ -56,7 +58,7 @@ def [](integration_name, configuration_name = :default)
           def instrument(integration_name, options = {}, &block)
             integration = fetch_integration(integration_name)
 
-            unless integration.nil?
+            unless integration.nil? || !integration_enabled?(integration_name)
```

Comment:
makes sense, i went ahead and took this approach and defined `enabled` in the shared class as well as each Contrib's setting's, and added the relevant Env var constants the respective `ext` files. 

- One thing is that it required me to modify some of the fixtures in the `extensions_spec`, and that i added a very small test in the `configurable_spec` to test the shared class. I found this pretty difficult to easily test at the individual contrib level, and i've punted on it in this PR for now. I think it will take some work done ala the `analytics_examples` / `analytics_spec` that abstracts everything out into shared helpers but the scope of that work could take a bit and didn't want it to be a blocker for this PR.